util/system/path.go

```diff
@@ -220,5 +221,30 @@ func CheckSystemDriveAndRemoveDriveLetter(path string, inputOS string) (string,
 	//
 	// We must return the second element of the split path, as is, without attempting to convert
 	// it to an absolute path. We have no knowledge of the CWD; that is treated elsewhere.
-	return ToSlash(filepath.Clean(parts[1]), inputOS), nil
+	return ToSlash(cleanPath(parts[1], inputOS, keepSlash), inputOS), nil
+}
+
+// An adaptation of filepath.Clean to allow an option to
+// retain the trailing slash, on either of the platforms.
+// See https://github.com/moby/buildkit/issues/5249
+func cleanPath(origPath, inputOS string, keepSlash bool) string {
+	// so as to handle cases like \\a\\b\\..\\c\\
+	// on Linux, when inputOS is Windows
+	origPath = ToSlash(origPath, inputOS)
+
+	if !keepSlash {
+		return filepath.Clean(origPath)
```

Comment:
I guess this should be `path.Clean` as `filepath` does not understand slash path in wcow.

Reply:
By this time, we all have `/` paths coz of `ToSlash(origPath, inputOS)` on L233 above.

Reply:
Yes, I undestand that, but `filepath` will not understand how to handle paths with `/` when running natively on windows. `path.Clean()` would understand paths with `/` on any platform.

Reply:
Oh, I get it, let me fix that. Thanks for the catch!